src/integrator/Adress.cpp

```diff
@@ -371,7 +381,14 @@ namespace espressopp {
         for (std::vector<Particle>::iterator it = adrATparticles.begin();
                 it != adrATparticles.end(); ++it) {
 
-            real dtfm = 0.5 * dt / it->mass();
+            real dtfm = 0.0;
+            if (afterSlowForces) {
```

Comment:
I do not think that it is necessary to duplicate the code for separate binds. However, you are right that the `if` inside the loop over the particles is implemented not in an optimal way. I pulled the `if` now out of the loop in the new commit.

Reply:
@kkreis  I think @MrTheodor suggestion regarding the duplicate method is practical, at least at this stage. Why? well, because we can have a special method containing your multi-time stepping PR, and after some performance tests will be performed by our collaborators in Garching and Co. We can be sure of its performance and naturally merge both into one. Otherwise this benchmarking will be required in advance and may take several months. Of course we can put a note in the current method that to use the multi-time stepping, you should refer to your specific method so that we have it available already for the users.